rero_mef/api_mef.py

```diff
@@ -51,33 +50,24 @@ def get_mef_by_entity_pid(cls, entity_pid, entity_name, pid_only=False):
         :returns: pid or record
         """
         key = f'{entity_name}.pid'
-        search = cls.search() \
+        if entity_name == 'viaf':
+            key = 'viaf_pid'
+        query = cls.search() \
             .filter('term', **{key: entity_pid}) \
-            .source(['pid'])
-        if search.count() > 1:
+            .params(preserve_order=True) \
+            .sort({'_updated': {'order': 'desc'}})
+        if pid_only:
+            mef_records = [hit.pid for hit in (query.source(['pid']).scan())]
+        else:
+            mef_records = [
+                cls.get_record_by_pid(hit.pid) for hit in
+                (query.source(['pid']).scan())
+            ]
```

Comment:
`get_record_by_pid`= 2 DB calls (1 for persistent identifier, 1 for record). Instead use :
```python
mef_records = [cls.get_record(hit.meta.id) for hit in query.scan()]
```
Bonus : less characters, maybe on a single line :-)